.github/workflows/ci-workflow.yml

```diff
@@ -0,0 +1,69 @@
+name: "CI Tests"
+on:
+  pull_request:
+    types: [assigned, opened, synchronize, reopened]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    services:
+      db:
+        image: postgres:11-alpine
+        ports: ['5432:5432']
+        env:
+          POSTGRES_PASSWORD: postgres
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        ref: ${{ github.head_ref }}
+
+    - uses: actions/setup-elixir@v1
+      with:
+        otp-version: '22.3.4.2'
+        elixir-version: '1.10.4'
+  
+    - name: Cache build
+      uses: actions/cache@v2
+      with: 
+        path: |
+          _build
+          deps
+        key: ${{ runner.os }}-mix-${{ hashFiles('**/mix.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-mix- 
+
+    - name: Install Dependencies
+      run: mix deps.get
+
+    - name: Compile
+      run: mix compile
```

Comment:
Should we run with the `--warnings-as-errors` option, so it makes sure that our code doesn't include any warning?

`mix compile --warnings-as-errors — Running this will return a non-zero exist status if your code contains any warnings.`

Reply:
If our code having some warning on compile then it will raise an error

![image](https://user-images.githubusercontent.com/11751745/90505545-fde41900-e17c-11ea-8768-ad708ad5d6b0.png)

Reply:
Updated 954fd18 🎉